pkg/http/api/v1/handler.go

```diff
@@ -15,11 +16,19 @@ import (
 
 type requestHandler struct {
 	adapter scanner.Adapter
+	logger Logger
 }
 
-func NewAPIHandler(adapter scanner.Adapter, logger io.Writer) http.Handler {
+type Logger interface {
```

Comment:
As the logger is not being used in any other package, I won't define a package for the logger interface. It is enough to provide a Logger satisfying this interface to the API handler.